src/main/java/org/glovo/glovo/service/OrderService.java

```diff
@@ -0,0 +1,109 @@
+package org.glovo.glovo.service;
+
+import lombok.AllArgsConstructor;
+import org.glovo.glovo.exception.OrderNotFoundException;
+import org.glovo.glovo.exception.OrderProcessingException;
+import org.glovo.glovo.exception.ProductNotFoundException;
+import org.glovo.glovo.model.Order;
+import org.glovo.glovo.model.Product;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class OrderService {
+
+    private final List<Order> orders;
+    private final ProductService productService;
+
+    public List<Order> getAllOrders() {
+        return orders;
+    }
+
+    public Order getOrderById(Long id) {
+        return orders.stream()
+                .filter(order -> order.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new OrderNotFoundException("Order not found with ID: " + id));
+    }
+
+    public Order createOrder(Order order) {
+        try {
+            order.setId(generateNextOrderId());
+            order.setCreationDateTime(LocalDateTime.now());
+            for (Product product : order.getProducts()) {
+                productService.addProduct(product);
+            }
+
+            orders.add(order);
+            recalculateOrderTotal(order);
+            return order;
```

Comment:
is it possible to create an order with products? if yes, can you please point me to the line with the logic